File: Java/Interface.java

interface UserModel {
  User login(User user);
  User create(User user);
}

class UserModelImp implements UserModel {

  @Override
  public User login(User user) {
    return user;
  }

  @Override
  public User create(User user) {
    return user;
  }
}

public class Interface {

  public static void main(String[] args) {
    var userModelImp = new UserModelImp();

    var lily = new User("lily", "12345678");
    var frank = new User("frank", "12345678");

    System.out.println(userModelImp.login(lily));
    System.out.println(userModelImp.create(frank));
  }
}

File: Kotlin/Interface.kt

interface UserModel {
  val prop: Int

  fun login (username: String, password: String): Boolean

  fun show () = println(prop)

  fun create () {}
}

interface People : UserModel {
  override val prop: Int get() = 50
}

class UserModelImp : UserModel, People {
  override val prop: Int get() = 20

  override fun login(username: String, password: String): Boolean {
    show().let {
      return !(username.isEmpty() || password.isEmpty())
    }
  }
}

fun main() {
  val userModelImp = UserModelImp()
  
  println(userModelImp.login("turaiiao", ""))
}